sdk/core/azure-core-serializer-json-jackson/src/main/java/com/azure/core/serializer/json/jackson/JacksonJsonSerializer.java

```diff
@@ -195,14 +238,41 @@ private String removePrefix(Method method) {
 
         AnnotatedClass annotatedClass = AnnotatedClassResolver.resolve(config,
             mapper.constructType(method.getDeclaringClass()), null);
-        AccessorNamingStrategy accessorNamingStrategy = config.getAccessorNaming().forPOJO(config, annotatedClass);
 
         AnnotatedMethod annotatedMethod = new AnnotatedMethod(null, method, null, null);
-        String name = accessorNamingStrategy.findNameForIsGetter(annotatedMethod, annotatedMethod.getName());
+        String annotatedMethodName = annotatedMethod.getName();
+
+        String name = null;
+        if (USE_REFLECTION_FOR_MEMBER_NAME) {
+            name = removePrefixWithReflection(config, annotatedClass, annotatedMethod, annotatedMethodName);
+        }
+
         if (name == null) {
-            name = accessorNamingStrategy.findNameForRegularGetter(annotatedMethod, annotatedMethod.getName());
+            name = removePrefixWithBeanUtils(annotatedMethod);
         }
 
         return name;
     }
+
+    private static String removePrefixWithReflection(MapperConfig<?> config, AnnotatedClass annotatedClass,
+        AnnotatedMethod method, String methodName) {
+        try {
+            Object accessorNamingStrategy = FOR_POJO.invoke(GET_ACCESSOR_NAMING.invoke(config), config, annotatedClass);
```

Comment:
These could result in NPE if FOR_POJO and other method handles are `null` (see above comment).